### PR TITLE
Merge develop into master

### PR DESCRIPTION
**Pull request by Aplicticus**

### CODE REVIEW

**Review by DanielvanBavel — DISMISSED** [review]

File: ContainerService.App/Messaging/ContainerMessageHandler.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			foreach (var container in receivedShip.Containers)
+			{
+				await _containerRepository.CreateContainerAsync(container);
+			}
+
+			await _containerRepository.UnloadContainersAsync(receivedShip.Containers);
+
+			await _messagePublisher.PublishMessageAsync(MessageTypes.ShipContainerUnloaded, receivedShip);
+
+			await _containerRepository.SortContainersAsync(receivedShip.Containers);
```
Comment (markieo1) [inline ContainerService.App/Messaging/ContainerMessageHandler.cs]: This only need to happen when a service is requested, for loading or unloading

File: ContainerService.Core/Models/Truck.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace ContainerService.Core.Models
+{
+	public class Truck
+	{
+		[Key]
+		public string LicensePlate { get; set; }
```
Comment (markieo1) [inline ContainerService.Core/Models/Truck.cs]: Truck can also contain a container. See TransportationService for this.

File: ContainerService.Infrastructure/Repositories/ContainerRepository.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		}
+
+		public async Task<List<Container>> SortContainersAsync(List<Container> containers)
+		{
+			return await Task.Run(() => containers);
+		}
+
+		public async Task<List<Container>> UnloadContainersAsync(List<Container> containers)
+		{
+			return await Task.Run(() => containers);
```
Comment (markieo1) [inline ContainerService.Infrastructure/Repositories/ContainerRepository.cs]: Unloading containers happens on available trucks

**Review by markieo1 — CHANGES_REQUESTED** [review]

File: ContainerService.Infrastructure/Repositories/ContainerRepository.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		}
+
+		public async Task<List<Container>> LoadContainersAsync(List<Container> containers)
+		{
+			return await Task.Run(() => containers);
+		}
+
+		public async Task<List<Container>> SortContainersAsync(List<Container> containers)
+		{
+			return await Task.Run(() => containers);
```
Comment (markieo1) [inline ContainerService.Infrastructure/Repositories/ContainerRepository.cs]: make sure to prioritize lifestock see: [order list](https://stackoverflow.com/questions/3309188/how-to-sort-a-listt-by-a-property-in-the-object)

**Review by markieo1 — CHANGES_REQUESTED** [review]

File: ContainerService.App/Messaging/ContainerMessageHandler.cs [hunk 5]
```diff
@@ -93,5 +86,45 @@ public class ContainerMessageHandler : IMessageHandlerCallback
...
+			var receivedShipService = JsonSerializer.Deserialize<ShipService>(message);
+			var trucks = await _truckRepository.GetTrucks();
+
+			var selectedTruck = trucks?.FirstOrDefault();
+
+			if (selectedTruck != null)
+			{
+				if (receivedShipService.Id == ShipServiceConstants.LoadContainerId)
+				{
+					await _containerRepository.LoadContainerAsync(selectedTruck.Container);
```
Comment (markieo1) [inline ContainerService.App/Messaging/ContainerMessageHandler.cs]: Don't we need to do this for every truck?

File: ContainerService.Core/Models/Container.cs [hunk 6]
```diff
@@ -30,5 +30,10 @@ public class Container
 		/// The type of the product.
 		/// </value>
 		public ProductType ProductType { get; set; }
+
+		/// <summary>
+		/// Gets or sets the container zone.
+		/// </summary>
+		public ContainerZone ContainerZone { get; set; }
```
Comment (markieo1) [inline ContainerService.Core/Models/Container.cs]: What is the usage of this?

**Review by markieo1 — CHANGES_REQUESTED** [review]

File: ContainerService.Infrastructure/Repositories/ContainerRepository.cs [hunk 7]
```diff
@@ -35,23 +36,11 @@ public ContainerRepository(ContainerDbContext containerDbContext)
...
-						container.ContainerZone = ContainerZone.LiveStockAndFresh;
-						break;
-					}
-				case ProductType.Normal:
-					{
-						container.ContainerZone = ContainerZone.Normal;
-						break;
-					}
-			}
+			containers.OrderBy(x => x.ProductType == ProductType.FreshProduceAndLivestock);
```
Comment (markieo1) [inline ContainerService.Infrastructure/Repositories/ContainerRepository.cs]: This won't work

File: ContainerService.Core/Models/ProductType.cs [hunk 8]
```diff
@@ -0,0 +1,8 @@
+﻿namespace ContainerService.Core.Models
+{
+	public enum ProductType
+	{
+		FreshProduceAndLivestock,
```
Comment (markieo1) [inline ContainerService.Core/Models/ProductType.cs]: Add = 1, or =2

File: ContainerService.Infrastructure/Repositories/ContainerRepository.cs [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+		}
+
+		public async Task<Container> LoadContainerAsync(Container container)
+		{
+			return await Task.Run(() => container);
+		}
+
+		public async Task<Container[]> SortContainersAsync(Container[] containers)
+		{
+			containers.OrderBy(x => x.ProductType == ProductType.FreshProduceAndLivestock);
```
Comment (markieo1) [inline ContainerService.Infrastructure/Repositories/ContainerRepository.cs]: This won't work, see: [C# Sort List by Enum
](https://stackoverflow.com/a/42552327/3563975)

**Review by markieo1 — CHANGES_REQUESTED** [review]

File: ContainerService.Infrastructure/Repositories/ContainerRepository.cs [hunk 10]
```diff
@@ -38,7 +38,7 @@ public ContainerRepository(ContainerDbContext containerDbContext)
 
 		public async Task<Container[]> SortContainersAsync(Container[] containers)
 		{
-			containers.OrderBy(x => x.ProductType == ProductType.FreshProduceAndLivestock);
+			containers.OrderByDescending(x => (int)(x.ProductType)).ToArray();
```
Comment (markieo1) [inline ContainerService.Infrastructure/Repositories/ContainerRepository.cs]: This should still be OrderBy, else we prioritize `ProductType.Normal`

**Review by markieo1 — CHANGES_REQUESTED** [review]

**Review by markieo1 — APPROVED** [review]

**Pull request merged**